### DISCUSSION TITLE
Allow File custom sensor to read multiple values

### DISCUSSION THREAD

**Discussion by nabsltd** [Rem0o/FanControl.Releases#2398] [Ideas]
Add the ability to read every line in the file and create multiple sensors with names like CustomName_01, CustomName_02, where "CustomName" is the name of the File sensor. Each line would be required to formatted just like the single line is now...a number that indicates the temperature in Celsius.

I created a script to grab the highest temperature of a set of sensors that are not supported by LHM. Since all these sensors are close together and covered by a single fan, this worked fine for me. This allowed me to only have to run the program that queried the hardware one time and get all the results, and select the max. Other use cases might want to have each individual value available.

Since writing to many files is slower, and it's more work to create multiple custom sensors, it would be nice to be able to just pull all the values in at once.

**Comment by Rem0o** [comment]
Hi,

While I think your request could be worked into a nice improvement for the file sensor, if you care at all about performance, did you look into the plugin route? The file sensor was created at the very beginning of FanControl as a *very* basic solution before the plugin system was added in as some sort of temporary solution to import external sensors.

https://github.com/Rem0o/FanControl.Releases/wiki/Plugins

>I created a script to grab the highest temperature of a set of sensors that are not supported by LHM.

You could import that directly into the software, in memory, without ever going through a file. 

Creating a plugin is pretty straight forward.   If you got VSCode or ideally Visual Studio community with the Dotnet SDK installed, it's a fun afternoon project. I could assist you if you need some guidance.
